### PR TITLE
ci: setup playwright

### PR DESCRIPTION
**Pull request by chintankavathia**
**What kind of change does this PR introduce?** (check one with "x")

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -118,7 +118,11 @@ const routes: Routes = [
 ];
 
 @NgModule({
-  imports: [RouterModule.forRoot(routes)],
+  imports: [
+    RouterModule.forRoot(routes, {
+      useHash: true
```
[the remark this answers is not in the document]
Comment (chintankavathia) [reply src/app/app-routing.module.ts:123]: Yes. otherwise routing doesn't work with playwright as it reloads the browser page on navigations

**Comment by spike-rabbit** [comment]
Just two things:
- please avoid at least API changes in this MR. Disabling checks should be fine
- use git lfs to store screenshots

File: .github/workflows/test_and_deploy.yml [hunk 2]
```diff
@@ -14,6 +14,12 @@ jobs:
         with:
           node-version: lts/iron
 
+      - name: Configure LFS Fetch URL
+        run: git config remote.origin.lfsurl https://github.com/siemens/ngx-datatable-snapshots.git/info/lfs
```
Comment (spike-rabbit) [inline .github/workflows/test_and_deploy.yml:18]: I think you can remove this, as we now have `.lfsconfig`

File: package.json [hunk 3]
```diff
@@ -16,7 +17,11 @@
     "test:ci": "ng test ngx-datatable-lib --watch=false --progress=false --browsers=ChromeHeadlessCI",
     "ci": "run-s lint test:ci",
     "lint": "ng lint",
-    "e2e": "ng e2e",
+    "e2e:prepare": "playwright install",
+    "lfs:prepare": "git config remote.origin.lfspushurl https://github.com/siemens/ngx-datatable-snapshots.git/info/lfs && git config remote.origin.lfsurl https://github.com/siemens/ngx-datatable-snapshots.git/info/lfs",
```
Comment (spike-rabbit) [inline package.json:21]: can be removed

File: package.json [hunk 4]
```diff
@@ -16,7 +17,11 @@
...
+    "vrt": "cross-env playwright test",
+    "vrt:local": "yarn vrt",
+    "vrt:update": "yarn vrt --update-snapshots",
```
Comment (spike-rabbit) [inline package.json:24]:
```suggestion
    "e2e": "playwright test",
    "e2e:update": "yarn e2e --update-snapshots",
```

File: package.json [hunk 5]
```diff
@@ -61,9 +66,11 @@
     "angular-cli-ghpages": "^0.6.0",
     "cpx": "^1.5.0",
     "cross-env": "^7.0.2",
+    "dotenv-cli": "^7.4.4",
```
Comment (spike-rabbit) [inline package.json:69]: this is not used, right?

File: playwright.config.ts [hunk 6]
```diff
@@ -0,0 +1,162 @@
+import { defineConfig, devices } from '@playwright/test';
+
+/**
+ * Read environment variables from file.
+ * https://github.com/motdotla/dotenv
+ */
+// import dotenv from 'dotenv';
+// import path from 'path';
+// dotenv.config({ path: path.resolve(__dirname, '.env') });
```
Comment (spike-rabbit) [inline playwright.config.ts:9]: please remove unused code

File: playwright.config.ts [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+    /* Test against branded browsers. */
+    // {
+    //   name: 'Microsoft Edge',
+    //   use: { ...devices['Desktop Edge'], channel: 'msedge' },
+    // },
+    // {
+    //   name: 'Google Chrome',
+    //   use: { ...devices['Desktop Chrome'], channel: 'chrome' },
+    // },
```
Comment (spike-rabbit) [inline playwright.config.ts:153]: please remove them as well

**Review by spike-rabbit — CHANGES_REQUESTED** [review]
Seems like we have some a11y issues which we should address in another MR soon.

**Review by spike-rabbit — APPROVED** [review]
thx a ton for the this. Should help us a lot to move on without breaking things.

**Pull request merged**